wallet/src/actors/app/methods.rs

```diff
@@ -574,6 +575,26 @@ impl App {
         Ok(())
     }
 
+    /// Handle node status notifications received from a Witnet node.
+    pub fn handle_node_status_notification(&mut self, value: types::Json) -> Result<()> {
+        let status = serde_json::from_value::<types::StateMachine>(value).map_err(node_error)?;
+
+        // This iterator is collected early so as to free the immutable reference to `self`.
+        let wallets: Vec<types::SessionWallet> = self
+            .state
+            .wallets
+            .iter()
+            .map(|(_, wallet)| wallet.clone())
+            .collect();
+
+        for wallet in wallets.iter() {
+            let sink = self.state.get_sink(&wallet.session_id);
+            self.handle_change_status_in_worker(&status, &wallet, sink.clone());
+        }
```

Comment:
I think you don't even need to `.iter()` in `for` statements, and in this case I think we can even iterate over a slice.
```suggestion
        for wallet in &wallets {
            let sink = self.state.get_sink(wallet.session_id);
            self.handle_change_status_in_worker(&status, wallet, sink.clone());
        }
```